### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by felixfbecker**
- Use files in package.json instead of .npmignore
- Don't publish tsconfig.json, prevents error in VS Code
- Publish src so debugging with source maps work
- Update TypeScript and emit declarationMaps
- Add webpack-cli to dependencies and fix webpack config (didn't build)

### CODE REVIEW

**Comment by felixfbecker** [comment]
@yurishkuro could you give this a review?

File: package.json [hunk 1]
```diff
@@ -4,6 +4,11 @@
   "engines": {
     "node": ">=0.10"
   },
+  "files": [
+    "src",
```
Comment (yurishkuro) [inline package.json]: why do we need src in the artifact?

File: package.json [hunk 2]
```diff
@@ -4,6 +4,11 @@
...
+    "lib",
```
Comment (yurishkuro) [inline package.json]: what is lib?

Comment (felixfbecker) [reply package.json]: It's the output dir of TypeScript, where compiled JavaScript goes.

https://github.com/opentracing/opentracing-javascript/blob/062505c9327c964c2925caaed9aa994ce5684f5e/tsconfig.json#L11

File: package.json (hunk 1 above)
Comment (felixfbecker) [reply package.json]: As mentioned in the PR description, without the source step debugging doesn't work as the source the sourcemaps point to cannot be found. The alternative would be inlining sources into the source maps (base64).

Comment (styfle) [reply package.json]: Another alternative is to avoid publishing both source and source maps.

File: package.json (hunk 2 above)
Comment (yurishkuro) [reply package.json]: how is that different from dist?

Comment (felixfbecker) [reply package.json]: `src` has `.ts` files. `dist` has `.js`, `.js.map`, `.d.ts` and `.d.ts.map` files (result of compiling `.ts` fields in `src`).
Note that that the only thing new here is generation of the `.d.ts.map` files and inclusion of `src` in the npm package.

You can compare https://unpkg.com/opentracing@0.14.3/dist/ to https://github.com/opentracing/opentracing-javascript/tree/master/src if you're curious about the differences of the files.

**Comment by yurishkuro** [comment]
would be good to have someone more versed in js / ts to have a look.

**Comment by austinlparker** [comment]
lgtm

**Comment by felixfbecker** [comment]
Since this was approved by two people, could this be merged?

I would really like to get rid of this logspam:

image

**Pull request merged**